src/ngx_postgres_util.c

```diff
@@ -61,10 +61,18 @@ ngx_postgres_upstream_finalize_request(ngx_http_request_t *r,
         u->resolved->ctx = NULL;
     }
 
+#if defined(nginx_version) && (nginx_version >= 1009001)
+    if (u->state && u->state->response_time) {
+#else
     if (u->state && u->state->response_sec) {
+#endif
         tp = ngx_timeofday();
+#if defined(nginx_version) && (nginx_version >= 1009001)
```

Comment:
So, if I understand you correct, it would, probably, be something like below, yes?
```
+#if defined(nginx_version) && (nginx_version >= 1009001)
+    if (u->state && u->state->response_time) {
         u->state->response_time = ngx_timeofday()->msec - u->state->response_time; // [!!!] or this line is unneeded entirely?
+#else
     if (u->state && u->state->response_sec) {
         tp = ngx_timeofday();
         u->state->response_sec = tp->sec - u->state->response_sec;
         u->state->response_msec = tp->msec - u->state->response_msec;
+#endif

<...>
```